lib/mini_scheduler/manager.rb

```diff
@@ -393,6 +389,77 @@ def self.discover_schedules
       schedules
     end
 
+    def self.hostname
+      @hostname ||=
+        begin
+          require "socket"
+          Socket.gethostname
+        rescue => e
+          begin
+            `hostname`.strip
+          rescue => e
+            "unknown_host"
+          end
+        end
+    end
+
+    # Discover running scheduled jobs on the current host.
+    #
+    # @example
+    #
+    #   MiniScheduler::Manager.discover_running_jobs
+    #
+    # @param hostname [String, nil] the hostname to filter jobs by. If nil, jobs acrosss all hosts are considered.
+    #
+    # @return [Array<Hash>] an array of hashes representing the running jobs.
+    # @option job [String] :job_class The class name of the job.
+    # @option job [Integer] :job_frequency_minutes The frequency of the job in minutes.
+    # @option job [Time] :job_started_at The time when the job started.
+    # @option job [String] :mini_scheduler_worker_thread_id The ID of the worker thread running the job.
+    #   The thread can be identified by matching the `:mini_scheduler_worker_thread_id` thread variable with the ID.
+    def self.discover_running_jobs
+      hostname = self.hostname
+
+      schedule_keys =
+        discover_schedules.reduce({}) do |acc, klass|
+          acc[klass] = if klass.is_per_host
+            self.schedule_key(klass, hostname)
+          else
+            self.schedule_key(klass)
+          end
+
+          acc
+        end
+
+      jobs = []
+
+      schedule_keys
+        .keys
+        .zip(MiniScheduler.redis.mget(*schedule_keys.values))
+        .each do |job_class, job_schedule_info|
+          next if job_schedule_info.nil?
+
+          parsed =
+            begin
+              JSON.parse(job_schedule_info, symbolize_names: true)
+            rescue JSON::ParserError
+              nil
+            end
+
+          next if parsed.nil?
+          next if parsed[:prev_result] != "RUNNING"
+
+          jobs << {
+            job_class: job_class.to_s,
+            job_frequency_minutes: job_class.every.in_minutes.to_i,
+            job_started_at: Time.at(parsed[:prev_run]),
+            mini_scheduler_worker_thread_id: parsed[:current_owner],
+          }
```

Comment:
@sam Instead of the above, do you mean something like the following?

```suggestion
          jobs << {
            job_class: job_class,
            job_started_at: Time.at(parsed[:prev_run]),
            mini_scheduler_worker_thread_id: parsed[:current_owner],
          }
```

Reply:
something like that... the internal terminology is all muddled though between schedule and job which makes it a bit confusing.